### PR TITLE
Add documentation for commands

### PR DESCRIPTION
**Pull request by viviicat**
- Also add the 'cmds' command for listing commands
- Add the 'help' command for both displaying command help, and displaying a quick starter message.

Relates to #6

### CODE REVIEW

File: src/common/cmd.c [hunk 1]
```diff
@@ -493,3 +544,20 @@ static cmdstat_t cmd_wait_fn(i32 argc, const char** argv)
...
+
+    u32* indices = StrDict_Sort(&ms_cmds, SDictStrCmp, NULL);
+
+    cmddef_t def = { 0 };
+    for (u32 i = 0; i < ms_cmds.count; i++)
+    {
+        u32 index = indices[i];
+        const char* key = ms_cmds.keys[index];
+        Con_Logf(LogSev_Info, "cmd", "\t%s", key);
+    }
```
Comment (viviicat) [inline src/common/cmd.c:560]: I was not sure how to iterate over the dict any other way. Is this the best way?

File: src/common/cmd.c [hunk 2]
```diff
@@ -493,3 +546,37 @@ static cmdstat_t cmd_wait_fn(i32 argc, const char** argv)
...
+        if ((i + 1) < argc)
+        {
+            StrCat(ARGS(text), " ");
+        }
+    }
+
+    Con_Logf(LogSev_Info, "cmd", text);
+
+    return cmdstat_ok;
+}
```
Comment (viviicat) [inline src/common/cmd.c:582]: This has the side effect of squashing `echo hello     world`  together, but original Quake does this too. Also, `echo "hello    world" preserves the whitespaces.

File: src/common/cmd.c (hunk 1 above)
Comment (gheshu) [reply src/common/cmd.c:562]: This is a good way to do it.
You can also loop over ms_cmds.keys by ms_cmds.width and check if ms_cmds.keys[i] is not null, but it won't be sorted.

**Comment by gheshu** [comment]
Looks good! Thanks a bunch!

**Pull request merged**